redaxo/src/addons/structure/plugins/content/pages/modules.modules.php

```diff
@@ -91,14 +92,13 @@
         }
 
         $error = rex_i18n::msg('module_cannot_be_deleted', $modulname);
-
-        if ('' != $moduleInUseMessage) {
-            $error .= '<ul>' . $moduleInUseMessage . '</ul>';
-        }
+        $error .= '<ul>' . $moduleInUseMessage . '</ul>';
     } else {
+        $del = rex_sql::factory();
```

Comment:
hier mit einem frischen rex_sql arbeiten, da phpstan sonst denkt, dass `$del->getRows() > 0` hier immer `true` ist, weil es weiter oben schonmal abgefragt wurde.

phpstan kann aber nicht realisieren, dass durch `setQuery` dass ergebnis von `getRows` ändert.

wir könnten `getRows` als impure markieren, dann würde es aber an anderen stellen fehler geben, wo es korrekt ist dass `getRows` gecached wird.
in meinen augen ist das wiederverwenden eines existierenden `rex_sql` durch weitere `setQuery` calls eher ein edge case und neue objekte zu verwenden kostet nix und erzeugt lokaleren code

Reply:
Wenn man eine Query mehrfach ausführen will, mit unterschiedlichen Params, dann ist es denke ich schon sinnvoll, am gleichen Objekt nur einmal prepare aufzurufen und dann mehrfach execute. In dem Fall ändert sich getRows dann ja auch.
Also mir kommt es nicht auf diese Stelle an. Hier können wir gerne neue Objekte erzeugen. Aber so generell gibt es denke ich schon valide Fälle mit sich änderndem getRows.

> wir könnten getRows als impure markieren, dann würde es aber an anderen stellen fehler geben, wo es korrekt ist dass getRows gecached wird.

Hast du hier ein Beispiel?

Reply:
siehe https://github.com/redaxo/redaxo/pull/5279

interessanterweise kommen gar keine neuen problem dazu, daher ggf. sogar der bessere weg.
dennoch könnte man hier neue objekte nehmen - finde ich in diesem fall trotzdem die bessere wahl.

aber dein beispiel war definitiv auch nützlich und macht mit #5279 noch mehr sinn